internal/commands/vaultsecrets/run/run.go

```diff
@@ -148,26 +149,55 @@ func getAllSecretsForEnv(opts *RunOpts) ([]string, error) {
 	}
 
 	result := os.Environ()
+	collisions := make(map[string]bool, 0)
 
 	for _, secret := range res.Payload.Secrets {
 		// we need to append results in case of duplicates we want secrets to override
 		switch {
 		case secret.RotatingVersion != nil:
 			for name, value := range secret.RotatingVersion.Values {
-				result = append(result, fmt.Sprintf("%v_%v=%v", strings.ToUpper(secret.Name), strings.ToUpper(name), value))
+				fmtName := fmt.Sprintf("%v_%v", strings.ToUpper(secret.Name), strings.ToUpper(name))
+				processCollisions(collisions, fmtName)
+				result = append(result, fmt.Sprintf("%v=%v", fmtName, value))
 			}
 		case secret.DynamicInstance != nil:
 			for name, value := range secret.DynamicInstance.Values {
-				result = append(result, fmt.Sprintf("%v_%v=%v", strings.ToUpper(secret.Name), strings.ToUpper(name), value))
+				fmtName := fmt.Sprintf("%v_%v", strings.ToUpper(secret.Name), strings.ToUpper(name))
+				processCollisions(collisions, fmtName)
+				result = append(result, fmt.Sprintf("%v=%v", fmtName, value))
 			}
 		case secret.StaticVersion != nil:
-			result = append(result, fmt.Sprintf("%v=%v", strings.ToUpper(secret.Name), secret.StaticVersion.Value))
+			fmtName := strings.ToUpper(secret.Name)
+			processCollisions(collisions, fmtName)
+			result = append(result, fmt.Sprintf("%v=%v", fmtName, secret.StaticVersion.Value))
+		}
+	}
+
+	// check collisions by dropping unique records from the map
+	for fmtName, collided := range collisions {
+		if !collided {
+			delete(collisions, fmtName)
+		} else {
+			_, err = fmt.Fprintf(opts.IO.Err(), "%s Environment variable \"%s\" was assigned more than once\n",
+				opts.IO.ColorScheme().WarningLabel(), fmtName)
+			if err != nil {
+				return nil, err
+			}
+
 		}
 	}
 
 	return result, nil
 }
 
+func processCollisions(collisions map[string]bool, fmtName string) {
```

Comment:
You could eliminate the conditional (and thereby the need for the func processCollisions) if you made collisions a map from string to int.  Then instead of doing `processCollisions(collisions, fmtName)`, you could just do `collisions[fmtName]++`.

Reply:
Used this as inspiration in 05baedf